src/Symfony/Component/Asset/Package.php

```diff
@@ -53,6 +58,21 @@ public function getUrl($path)
     }
 
     /**
+     * {@inheritdoc}
+     */
+    public function getAndPreloadUrl($path, $as = '', $nopush = false)
+    {
+        if (null === $this->preloadManager) {
+            throw new LogicException('There is no preload manager, configure one first.');
+        }
+
+        $url = $this->getUrl($path);
+        $this->preloadManager->addResource($url, $as, $nopush);
```

Comment:
should the package really be the part being aware of the PreloadManager ? I don't think so. IMO, it should be done at a higher level in the stack. The preloading part is totally independent from the asset url building anyway